src/Elrond/Codec.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Codec.h"
+
+#include "HexCoding.h"
+
+using namespace TW;
+using namespace TW::Elrond;
+
+void normalizeHexStringTopLevel(std::string& value);
+
+std::string Codec::encodeStringTopLevel(const std::string& value) {
```

Comment:
maybe we can remove suffix like `TopLevel` from all methods?

Reply:
We have two distinct types of encoding (top level vs. nested) - however, indeed, the suffix was redundant in this case, I've removed it. In the future, if needed (most probably not), we might define two distinct classes: `TopLevelCodec` and `NestedCodec`.